### PR TITLE
Engine plugin changes

### PR DESCRIPTION
**Pull request by anmavrid**

### CODE REVIEW

File: src/plugins/JavaBIPEngine/JavaBIPEngine.js [hunk 1]
```diff
@@ -124,31 +159,66 @@ define([
...
+                                compileCode += 'javac -cp "' + path + '/:' + process.cwd() + '/engineLibraries/*" ' + path + '/' + fileName;
 
+                                fs.writeFileSync(path + '/compile.sh', compileCode, 'utf8');
+                                self.sendNotification('Compilation script has been successfully created.');
 
+                                child = execSync('chmod 775 ' + path + '/compile.sh');
+                                try {
+                                    child = execSync(path + '/compile.sh');
+                                } catch (e) {
+                                    console.log(e.stderr);
```
Comment (pmeijer) [inline src/plugins/JavaBIPEngine/JavaBIPEngine.js]: The execution must stop here

File: src/plugins/JavaBIPEngine/JavaBIPEngine.js [hunk 2]
```diff
@@ -124,31 +159,66 @@ define([
...
+                                simulateCode = 'java -cp "' + path + '/:' + process.cwd() + '/engineLibraries/*" org.junit.runner.JUnitCore ' + fileName.slice(0, -5);
+
+                                fs.writeFileSync(path + '/simulate.sh', simulateCode, 'utf8');
+                                self.sendNotification('Simulation script has been successfully created.');
+
+                                child = execSync('chmod 775 ' + path + '/simulate.sh');
+                                try {
+                                    child = execSync(path + '/simulate.sh');
+                                } catch (e) {
+                                    console.log('stderr ' + e.stderr);
```
Comment (pmeijer) [inline src/plugins/JavaBIPEngine/JavaBIPEngine.js]: Here as well

File: src/plugins/JavaBIPEngine/JavaBIPEngine.js [hunk 3]
```diff
@@ -124,31 +159,66 @@ define([
...
-                                //     });
-                                // fs.writeFileSync(path + '/' + engineOutputFileName, filesToAdd[engineOutputFileName], 'utf8');
-                            }
+                                for (file in behaviorFiles) {
+                                    compileCode += 'javac -cp "' + process.cwd() + '/engineLibraries/*" ' + path + '/' + file + '\n\n';
+                                }
+                                compileCode += 'javac -cp "' + path + '/:' + process.cwd() + '/engineLibraries/*" ' + path + '/' + fileName;
 
+                                fs.writeFileSync(path + '/compile.sh', compileCode, 'utf8');
+                                self.sendNotification('Compilation script has been successfully created.');
```
Comment (pmeijer) [inline src/plugins/JavaBIPEngine/JavaBIPEngine.js]: I would change this to:
`Compiling java code..`

`Compilation successful.`

`Calling simulate..`

`Simulate successful.`

Or something like that.

Comment (anmavrid) [inline src/plugins/JavaBIPEngine/JavaBIPEngine.js]: Done.

**Pull request merged**